test/utils/base.py

```diff
@@ -232,36 +232,26 @@ def setUpClass(cls):
         # Safe to remove once https://github.com/Qiskit/qiskit-aer/pull/2179 is in a release version
         # of Aer.
         warnings.filterwarnings(
-            "default",
+            "ignore",
             category=DeprecationWarning,
             message="Treating CircuitInstruction as an iterable is deprecated",
             module=r"qiskit_aer(\.[a-zA-Z0-9_]+)*",
         )
 
+        # Add things here carefully, specially if they are *broad*. If they get forgotten here,
+        # they might silence important things. So, add a comment for each of them. On why they are there
+        # and when can they be removed.
+        #  Ideally, use
+        # `allow_DeprecationWarning_message` instead, because it can be much more specific.
         allow_DeprecationWarning_modules = [
-            "test.python.pulse.test_builder",
-            "test.python.pulse.test_block",
-            "test.python.quantum_info.operators.symplectic.test_legacy_pauli",
-            "qiskit.quantum_info.operators.pauli",
-            "pybobyqa",
-            "numba",
-            "qiskit.utils.measurement_error_mitigation",
-            "qiskit.circuit.library.standard_gates.x",
-            "qiskit.pulse.schedule",
-            "qiskit.pulse.instructions.instruction",
-            "qiskit.pulse.instructions.play",
-            "qiskit.pulse.library.parametric_pulses",
-            "qiskit.quantum_info.operators.symplectic.pauli",
+            # Example - Because the deprecation of X #<issue/PR>. Can be removed once X is removed.
+            # "qiskit.quantum_info.operators.symplectic.pauli",
         ]
         for mod in allow_DeprecationWarning_modules:
             warnings.filterwarnings("default", category=DeprecationWarning, module=mod)
```

Comment:
We might consider removing this entirely; there's very little reason to ever blanket-allow warnings from an entire module, so imo we shouldn't make it easy to do like this.  The examples above with explicit and very tight warning filters are better.

Reply:
done in cb1d6ef170ae85cd1399410d34f0bacb2f753a7e